course_discovery/apps/course_metadata/management/commands/refresh_course_metadata.py

```diff
@@ -62,136 +89,151 @@ def add_arguments(self, parser):
+            dest='maintenance_period',
+            default=None,
+            help='maintenance period for some special tasks.  (e.g: sync. all of courses from LMS) Argument format: "10~16"'
+        )
 
     def handle(self, *args, **options):
-        # We only want to invalidate the API response cache once data loading
-        # completes. Disconnecting the api_change_receiver function from post_save
-        # and post_delete signals prevents model changes during data loading from
-        # repeatedly invalidating the cache.
-        for model in apps.get_app_config('course_metadata').get_models():
-            for signal in (post_save, post_delete):
-                signal.disconnect(receiver=api_change_receiver, sender=model)
-
-        # For each partner defined...
-        partners = Partner.objects.all()
-
-        # If a specific partner was indicated, filter down the set
-        partner_code = options.get('partner_code')
-        if partner_code:
-            partners = partners.filter(short_code=partner_code)
-
-        if not partners:
-            raise CommandError('No partners available!')
-
-        token_type = 'JWT'
-        for partner in partners:
-            logger.info('Retrieving access token for partner [{}]'.format(partner.short_code))
-
-            try:
-                access_token, __ = EdxRestApiClient.get_oauth_access_token(
-                    '{root}/access_token'.format(root=partner.oidc_url_root.strip('/')),
-                    partner.oidc_key,
-                    partner.oidc_secret,
-                    token_type=token_type
+        with _ScriptLock() as _script_locker:    # It make sure Only one script running in a time.
+
+            # We only want to invalidate the API response cache once data loading
+            # completes. Disconnecting the api_change_receiver function from post_save
+            # and post_delete signals prevents model changes during data loading from
+            # repeatedly invalidating the cache.
+            for model in apps.get_app_config('course_metadata').get_models():
+                for signal in (post_save, post_delete):
+                    signal.disconnect(receiver=api_change_receiver, sender=model)
+
+            # For each partner defined...
+            partners = Partner.objects.all()
+
+            # If a specific partner was indicated, filter down the set
+            partner_code = options.get('partner_code')
+            if partner_code:
+                partners = partners.filter(short_code=partner_code)
+
+            if not partners:
+                raise CommandError('No partners available!')
+
+            # parse maintenance period. script can do some special tasks in this period
+            maintenance_period = options.get('maintenance_period')
+            maintenance_period = [int(h) for h in maintenance_period.split('~')] if maintenance_period and '~' in maintenance_period else []
+            maintenance_period.sort()
```

Comment:
```
Usage: 
python manage.py refresh_course_metadata --maintenance_period=5~7
```

IF specify argument `--maintenance_period` with value `5~7`, that means:
 - While running between Am5 ~ Am7. We sync. all of courses from LMS
 - Otherwise we only sync. courses edited in 24hrs only.